orbit/utils/params_tuning.py

```diff
@@ -117,30 +130,36 @@ def _get_params(model):
                     "tuned hyper-param {} is not in the model's parameters".format(key)
                 )
 
-        # it is safer to reinstantiate a model object than using deepcopy...
+        # it is safer to re-instantiate a model object than using deepcopy...
         new_model_template = model._model.__class__(**params_tmpl_)
         new_model = model.__class__(model=new_model_template, **params_)
 
-        bt = BackTester(
-            model=new_model,
-            df=df,
-            min_train_len=min_train_len,
-            n_splits=n_splits,
-            incremental_len=incremental_len,
-            forecast_len=forecast_len,
-            **kwargs,
-        )
-        bt.fit_predict()
-        # TODO: should we assert len(metrics) == 1?
-        if metrics is None:
-            metrics = smape
-        metric_val = bt.score(metrics=[metrics]).metric_values[0]
-        if verbose:
-            logger.info("tuning metric:{:-.5g}".format(metric_val))
-        metric_values.append(metric_val)
+        if eval_method == "backtest":
+            bt = BackTester(
+                model=new_model,
+                df=df,
+                min_train_len=min_train_len,
+                n_splits=n_splits,
+                incremental_len=incremental_len,
+                forecast_len=forecast_len,
+                **kwargs,
+            )
+            bt.fit_predict()
+            # TODO: should we assert len(metrics) == 1?
+            if metrics is None:
+                metrics = smape
+            metric_val = bt.score(metrics=[metrics]).metric_values[0]
+            if verbose:
+                logger.info("tuning metric:{:-.5g}".format(metric_val))
+            metric_values.append(metric_val)
+        elif eval_method == "bic":
+            new_model.fit(df)
+            metric_values.append(new_model.get_bic())
```

Comment:
so, the 'bic' criteria only supports MAP estimator? when it is MCMC, this will pop an error?

Reply:
what's your suggestion? shall we put exception here? or adding wbic after checking the object type?